### PR TITLE
Issue 3745: Moved security constants

### LINKED ISSUE #3745: Move security constants to :shared:test
**Problem description**
According to the conversation in this thread of #3693:

https://github.com/pravega/pravega/pull/3693#discussion_r279546589

we want to move security constants in: 

test/testcommon/src/main/java/io/pravega/test/common/SecurityConfigDefaults.java

to :shared:test, which is a package that doesn't exist currently. These defaults are specific to Pravega and as such shouldn't be in `:test:common`. `:test:common` is for utility classes that can be used outside the scope of Pravega, while `:shared` is where we have shared classes that are useful only in the scope of Pravega.

**Problem location**
Security defaults for testing purposes.

**Suggestions for an improvement**
Make the necessary gradle changes to accomodate `:shared:test` and move `SecurityConfigDefaults` there.

### PR DESCRIPTION
**Pull request by Prabhaker24**
**Change log description**  
moved the security constants from ```:test:common``` to ```:shared:security```

**Purpose of the change**  
fixes #3745 

**What the code does**  
Moved SecurityConfigDefaults from ```:test:common``` to ```:shared:security``` as these defaults are specific to pravega and shouldn't be inside the ```:test:common```, many packages had test dependency on the ```shared: security``` i.e the tests in many packages depended on tests of ```shared: security``` and the best way of including such dependency is to create a jarTest for ```shared:security``` jarTest task builds the jar from the test outputs adding a ‘test’ to the jar name then I have created a configuration which basically is a group of dependencies and finally added the jar artifact to the configuration I have passed the configuration as parameter to the project that depends on it , thus whenever someone(project which is dependent on it) ask for testOutput, gradle will run the jarTest task

**How to verify it**  
Running all the tests with tls enabled

### CODE REVIEW

File: shared/security/src/test/java/io/pravega/shared/security/utils/SecurityConfigDefaults.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
  *
  *     http://www.apache.org/licenses/LICENSE-2.0
  */
-package io.pravega.test.common;
+package io.pravega.shared.security.utils;
```
Comment (ravisharda) [inline shared/security/src/test/java/io/pravega/shared/security/utils/SecurityConfigDefaults.java]: It isn't clear looking at the package name or the class name (`io.pravega.shared.security.utils.SecurityConfigDefaults`) that this class is for use in test classes only, even though the location of the source file clarifies it a bit. I'd suggest that it should be made explicit by relocating this class to: `io.pravega.shared.security.test.SecurityConfigDefaults`.

File: build.gradle [hunk 2]
```diff
@@ -169,6 +169,19 @@ project('shared:authplugin') {
 }
 
 project('shared:security') {
+    task jarTest (type: Jar) {
```
Comment (ravisharda) [inline build.gradle]: Can you explain why these changes are needed in the PR `description > what the code does?`?

File: build.gradle [hunk 3]
```diff
@@ -249,6 +262,7 @@ project('client') {
         // The client fallsback to NIO based transport if epoll is not available.
         compile group: 'io.netty', name: 'netty-transport-native-epoll', version: nettyVersion, classifier: 'linux-x86_64'
         compile group: 'com.google.guava', name: 'guava', version: guavaVersion
+        testCompile project(path: ':shared:security', configuration: 'testOutput')
```
Comment (ravisharda) [inline build.gradle]: Why was this needed?

File: build.gradle [hunk 4]
```diff
@@ -469,6 +484,7 @@ project('test:integration') {
         compile project(':test:testcommon')
         compile group: 'junit', name:'junit', version: junitVersion
         compile group: 'org.apache.curator', name: 'curator-test', version: apacheCuratorVersion
+        compile project(path: ':shared:security', configuration: 'testOutput')
```
Comment (ravisharda) [inline build.gradle]: Why is this needed, and what is it doing?

File: shared/security/src/test/java/io/pravega/shared/security/utils/SecurityConfigDefaults.java (hunk 1 above)
Comment (Prabhaker24) [reply shared/security/src/test/java/io/pravega/shared/security/utils/SecurityConfigDefaults.java]: I have changed the package name

File: build.gradle (hunk 2 above)
Comment (Prabhaker24) [reply build.gradle]: many packages had test dependency on the ```shared: security``` i.e the tests in many packages depended on tests of ```shared: security``` and the best way of including such dependency is to create a jarTest for ```shared:security``` jarTest task builds the jar from the test outputs adding a ‘test’ to the jar name then I have created a configuration which basically is a group of dependencies and finally added the jar artifact to the configuration I have passed the configuration as parameter to the project that depends on it , thus whenever someone(project which is dependent on it) ask for testOutput, gradle will run the jarTest task

File: build.gradle (hunk 3 above)
Comment (Prabhaker24) [reply build.gradle]: this line is needed as project client has test dependency over project security, when this is run gradle will run the jarTest task in the security project in the build.gradle.

File: build.gradle (hunk 4 above)
Comment (Prabhaker24) [reply build.gradle]: This line is needed as the project integration has compile dependency over the test present in the project security so when this is run gradle will run the jarTest task in the security project in the build.gradle.

**Comment by andreipaduroiu** [comment]
@Prabhaker24 Are we still pursuing this PR?

**Comment by Prabhaker24** [comment]
> @Prabhaker24 Are we still pursuing this PR?

@ravisharda can you please take a look at the pr again?

**Comment by andreipaduroiu** [comment]
@ravisharda @Prabhaker24 Please let me know your intentions with this PR. If I don't see any response or activity by March 31, I will close it as being abandoned.

**Comment by andreipaduroiu** [comment]
Closed for inactivity. Please reopen if work is to resume on this task.
